v2/processor_test.go

```diff
@@ -0,0 +1,173 @@
+package v2_test
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azservicebus"
+
+	"github.com/stretchr/testify/require"
+
+	v2 "github.com/Azure/go-shuttle/v2"
+)
+
+func MyHandler(timePerMessage time.Duration) v2.HandlerFunc {
+	return func(ctx context.Context, settler v2.MessageSettler, message *azservicebus.ReceivedMessage) {
+		// logic
+		time.Sleep(timePerMessage)
+		err := settler.CompleteMessage(ctx, message, nil)
+		if err != nil {
+			panic(err)
+		}
+	}
+}
+
+func ExampleProcessor() {
+	tokenCredential, err := azidentity.NewDefaultAzureCredential(nil)
```

Comment:
I think this is causing the Azure pipeline to fail because it requires interactive login?